foundation/plugins/org.jboss.tools.foundation.ui/src/org/jboss/tools/foundation/ui/widget/WidgetVisitorUtility.java

```diff
@@ -0,0 +1,148 @@
+			@Override
+			public boolean visit(Control control) {
+				control.setEnabled(enabled);
+				return true;
+			}
+		}, composite);
+	}
+	
+	/**
+	 * Functionally equivalent to accept(visitor, composite, false, false)
+	 * for a non-depth-first recursive call that ignores the parent composite itself. 
+	 * @param visitor
+	 * @param composite
+	 */
+	public void doForAllChildren(IWidgetVisitor visitor, Composite composite) {
+		accept(visitor, composite, false, false);
+	}
+	
+	/**
+	 * Accepts the given visitor.
+	 * 
+	 * The visitor's <code>visit</code> method is called on the given composite first.
+	 *  
+	 * The list of children for the given composite will then be visited.
+	 * If the <code>visit</code> method returns true for a given child
+	 * widget, and that widget is also a composite, the <code>accept</code>
+	 * method will be called on all children of that composite as well.   
+	 * 
+	 * This is a shortcut accept(visitor, composite, false)
+	 * 
+	 * @param visitor     The visitor
+	 * @param composite   The composite to visit
+	 */
+	public void accept(IWidgetVisitor visitor, Composite composite) {
+		accept(visitor, composite, false);
+	}
+	
+	/**
+	 * Accepts the given visitor.
+	 * 
+	 * The visitor's <code>visit</code> method is called on the given composite first, 
+	 * or last, depending on the value of <code>depthFirst</code>. 
+	 *  
+	 * The list of children for the given composite will then be visited.
+	 * If the <code>visit</code> method returns true for a given child
+	 * widget, and that widget is also a composite, the <code>accept</code>
+	 * method will be called on all children of that composite as well.   
+	 * 
+	 * This is a shortcut accept(visitor, composite, depthFirst, true)
+	 * 
+	 * @param visitor     The visitor
+	 * @param composite   The composite to visit
+	 * @param depthFirst  Whether to perform depth first or not
+	 */
+	public void accept(IWidgetVisitor visitor, Composite composite, boolean depthFirst) {
+		accept(visitor, composite, depthFirst, true);
+	}
+	
+	/**
+	 * Accepts the given visitor.
+	 * 
+	 * The visitor's <code>visit</code> method is called on the given composite first, 
+	 * or last, depending on the value of <code>depthFirst</code>. This will not occur
+	 * if <code>visitParent</code> is set to <code>false</code>
+	 *  
+	 * The list of children for the given composite will then be visited.
+	 * If the <code>visit</code> method returns true for a given child
+	 * widget, and that widget is also a composite, the <code>accept</code>
+	 * method will be called on all children of that composite as well.   
+	 * 
+	 * @param visitor     The visitor
+	 * @param composite   The composite to visit
+	 * @param depthFirst  Whether to perform depth first or not
+	 * @param visitParent Whether to visit the parent or to proceed directly to the children
+	 */
+	public void accept(IWidgetVisitor visitor, Composite composite, boolean depthFirst, boolean visitParent) {
```

Comment:
I'd transform these depthFirst and visitParent parameters to fields of WidgetVisitorUtility